app/views/shared/_language_selector.html.erb

```diff
@@ -0,0 +1,16 @@
+<label class="position-relative d-flex <%= classes %>" role="button">
+  <svg width="20" height="20" fill="currentColor">
+    <path d="M20 18h-1.44a.61.61 0 0 1-.4-.12.81.81 0 0 1-.23-.31L17 15h-5l-1 2.54a.77.77 0 0 1-.22.3.59.59 0 0 1-.4.14H9l4.55-11.47h1.89zm-3.53-4.31L14.89 9.5a11.62 11.62 0 0 1-.39-1.24q-.09.37-.19.69l-.19.56-1.58 4.19zm-6.3-1.58a13.43 13.43 0 0 1-2.91-1.41 11.46 11.46 0 0 0 2.81-5.37H12V4H7.31a4 4 0 0 0-.2-.56C6.87 2.79 6.6 2 6.6 2l-1.47.5s.4.89.6 1.5H0v1.33h2.15A11.23 11.23 0 0 0 5 10.7a17.19 17.19 0 0 1-5 2.1q.56.82.87 1.38a23.28 23.28 0 0 0 5.22-2.51 15.64 15.64 0 0 0 3.56 1.77zM3.63 5.33h4.91a8.11 8.11 0 0 1-2.45 4.45 9.11 9.11 0 0 1-2.46-4.45z" />
+  </svg>
+  <% unless disabled %>
+    <select role="button" class="p-0 position-absolute top-0 start-0 w-100 h-100 language-change-trigger text-transparent bg-transparent <%= classes %>">
+      <% Locale.available
+               .map { |locale| Language.find_by(:code => locale.to_s) }
+               .select { |locale| locale }
+               .sort_by { |locale| locale[:english_name] }
+               .each do |language| %>
+        <option class="form-select" value="<%= language.code %>" <%= "selected" if I18n.locale.to_s == language.code %>><%= language.name %></option>
```

Comment:
Using Language table is probably a bad idea. I tried this:

```suggestion
      <% Locale.available
               .select { |locale| I18n.exists? "shared.language_selector.language", :locale => locale, :fallback => false }
               .sort_by { |locale| locale.to_s == "en" ? "English" : t(".language", :locale => locale) }
               .each do |locale| %>
        <option class="form-select" value="<%= locale.to_s %>" <%= "selected" if I18n.locale.to_s == locale.to_s %>><%= locale.to_s == "en" ? "English" : t(".language", :locale => locale) %></option>
```

with this in locale file:

```
  share:
    language_selector:
      language: THIS LANGUAGE NAME
```

Now translators have to add whatever is the appropriate representation of their language there, and then it will appear in the selector.

Reply:
Pushed code can be shortened as `Language.where(:code => Locale.available.map {|locale| locale.to_s})`. This approach will also cause only one DB query. If there is no other reason for language not to be used, I'll push the code with this approach. Translating language names using i18n and not having one source for them, may cause duplicate translations or inconsistency (because of multiple functionalities using language names).

Reply:
> Translating language names using i18n and not having one source for them

i18n is going to be their *one source*

> may cause duplicate translations or inconsistency (because of multiple functionalities using language names).

We already have two different functionalities using two different sets of languages: i18n and languages of diary entries.